lib/charms/opensearch/v0/opensearch_backups.py

```diff
@@ -0,0 +1,641 @@
+        except OpenSearchKeystoreError:
+            # Ignore the delete error, as it may mean there was a certificate at a point
+            # and the user removed it from the configuration later on.
+            pass
+
+        try:
+            if self.charm.plugin_manager.run():
+                self.charm.on[self.charm.service_manager.name].acquire_lock.emit(
+                    callback_override="_restart_opensearch"
+                )
+        except OpenSearchError as e:
+            if isinstance(e, OpenSearchPluginRelationClusterNotReadyError):
+                self.charm.unit.status = WaitingStatus("s3-depart event: cluster not ready yet")
+            else:
+                self.charm.unit.status = BlockedStatus(
+                    "Unexpected error during plugin configuration, check the logs"
+                )
+                # There was an unexpected error, log it and block the unit
+                logger.error(e)
+            event.defer()
+            return
+        self.charm.unit.status = ActiveStatus()
+
+    def _execute_s3_depart_calls(self):
+        """Executes the s3 departure API calls."""
+        return  # do not execute anything as we intend to keep the backups untouched
+
+    def _check_repo_status(self) -> BackupServiceState:
+        try:
+            return self.get_service_status(
+                self._request("GET", f"_snapshot/{OPENSEARCH_REPOSITORY_NAME}")
+            )
+        except OpenSearchBackupNeworkError:
+            return BackupServiceState.RESPONSE_FAILED_NETWORK
+
+    def _check_snapshot_status(self) -> BackupServiceState:
+        try:
+            return self.get_snapshot_status(self._request("GET", "/_snapshot/_status"))
+        except OpenSearchBackupNeworkError:
+            return BackupServiceState.RESPONSE_FAILED_NETWORK
+
+    def _service_already_registered(self, bucket_name: str = "") -> bool:
+        """Returns True if the snapshot repo has already been created.
+
+        If bucket_name is set, then compare it in the response as well.
+        """
+        state = self._check_repo_status()
+        if state == BackupServiceState.SUCCESS:
+            # Repo already present
+            return True
+        return False
+
+    def _register_snapshot_repo(self) -> BackupServiceState:
+        """Registers the snapshot repo in the cluster."""
+        info = self.s3_client.get_s3_connection_info()
+        bucket_name = info["bucket"]
+        return self.get_service_status(
+            self._request(
+                "PUT",
+                f"_snapshot/{OPENSEARCH_REPOSITORY_NAME}",
+                payload={
+                    "type": "s3",
+                    "settings": {
+                        "bucket": bucket_name,
+                        "base_path": info.get("path", S3_REPO_BASE_PATH),
+                    },
+                },
+            )
+        )
+
+    def _check_missing_s3_config_completeness(self) -> List[str]:
```

Comment:
Instead of replacing this method to a boolean, I am actually adding more information in both log and status message containing what is actually missing